### PR TITLE
implement module

### PR DESCRIPTION
**Pull request by Qard**
This is an initial implementation of a set-cookie parser and serializer.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+const encode = Symbol('encode')
+
+function parsePair (segment) {
+  return segment.trim().split('=')
+}
+
+class SetCookie {
+  constructor (input, options) {
+    if (Array.isArray(input)) {
+      return input.map(item => new SetCookie(item, options))
```
Comment (watson) [inline index.js:17]: Will this work if people do `new SetCookie(array)`?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      }
+
+      data = {
+        key: this[decode](pair[0]),
+        value: this[decode](pair[1]),
+        expires: undefined,
+        maxAge: undefined,
+        domain: undefined,
+        path: undefined,
+        secure: undefined,
```
Comment (watson) [inline index.js]: Consider setting this default value to `false`

File: index.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        httpOnly: undefined,
```
Comment (watson) [inline index.js]: Consider setting this default value to `false`

File: index.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    if (typeof input === 'string') {
+      const segments = input.split(';').map(parsePair)
+      const pair = segments.shift()
+
+      if (pair.length < 2 || !pair[0] || !pair[1]) {
+        throw new Error('Invalid key-value pair')
+      }
+
+      data = {
+        key: this[decode](pair[0]),
```
Comment (watson) [inline index.js]: From https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie:

- `__Secure-` prefix: Cookies with a name starting with `__Secure-` (dash is part of the prefix) must be set with the `secure` flag and must be from a secure page (HTTPS).
- `__Host-` prefix: Cookies with a name starting with `__Host-` must be set with the `secure` flag, must be from a secure page (HTTPS), must not have a domain specified (and therefore aren't sent to subdomains) and the path must be "/".

**Review by watson — COMMENTED** [review]
Looks good! Left a few comments. Btw, when adding `.travis.yml`, I recommend adding Node.js v4 and v9 to the matrix to match Elastic APM

File: index.js (hunk 1 above)
Comment (Qard) [reply index.js:17]: Yep, it will return an array of SetCookie instances.

Comment (watson) [reply index.js:17]: Will return statements not be ignored inside of a constructor?

File: index.js (hunk 4 above)
Comment (watson) [reply index.js]: Did you consider this?

File: index.js (hunk 1 above)
Comment (Qard) [reply index.js:17]: Nope, if you return in a constructor, the return value of the `new Constructor` will be whatever was returned. 😸

Comment (Qard) [reply index.js:17]: The tests prove this already, by the way. 😸

**Comment by Qard** [comment]
@watson Can I get another review of this?

**Review by watson — APPROVED** [review]

**Pull request merged**